src/components/views/dialogs/DMInviteDialog.js

```diff
@@ -48,8 +74,46 @@ class DMRoomTile extends React.PureComponent {
         this.props.onToggle(this.props.member.userId);
     };
 
+    _highlightName(str: string) {
+        if (!this.props.highlightWord) return str;
+
+        // First encode the thing to avoid injection
+        str = htmlEntitiesEncode(str);
+
+        // We convert things to lowercase for index searching, but pull substrings from
+        // the submitted text to preserve case.
+        const lowerStr = str.toLowerCase();
+        const filterStr = this.props.highlightWord.toLowerCase();
+
+        const result = [];
+
+        let i = 0;
+        let ii;
+        while ((ii = lowerStr.indexOf(filterStr, i)) >= 0) {
+            // Push any text we missed (first bit/middle of text)
+            if (ii > i) {
+                // Push any text we aren't highlighting (middle of text match)
+                result.push(<span key={i + 'mid'}>{str.substring(i, ii)}</span>);
```

Comment:
Would `begin` be more logical and parallel `end` below, since it's before the highlighted text?

Reply:
It's at the beginning or middle (for instance, searching for 'a' in 'Charlie Andrews' will give a segment of 'Ch', 'a', 'rlie ', 'A', and 'ndrews'. `mid` is supposed to cover 'Ch' and 'rlie '.